speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,483 @@
+"""This lobe enables the integration of huggingface pretrained SpeechT5 models for Automatic Speech Recognition.
+Reference: https://aclanthology.org/2022.acl-long.393.pdf
+
+Transformer from HuggingFace needs to be installed:
+https://huggingface.co/transformers/installation.html
+
+Authors
+ * Haroun Elleuch 2024
+"""
+
+import gc
+import logging
+from typing import Tuple
+
+import torch
+from transformers import (
+    GenerationConfig,
+    SpeechT5Config,
+    SpeechT5ForSpeechToText,
+)
+
+from speechbrain.lobes.models.huggingface_transformers.huggingface import (
+    HFTransformersInterface,
+)
+from speechbrain.utils.checkpoints import (
+    mark_as_loader,
+    mark_as_saver,
+    register_checkpoint_hooks,
+)
+from speechbrain.utils.fetching import fetch
+
+logger = logging.getLogger(__name__)
+
+
+@register_checkpoint_hooks
```

Comment:
Do you remember why this line is necessary ? It seems to be the only HF interface that is using this checkpoint hook.

Reply:
Yes, there was an issue where the model would not be saved correctly, and a shape mismatch would occur when resuming training from a checkpoint created during training
However, loading a model from HF when starting the training and saving the resulting checkpoint did not result in errors.
That's why I had to use a custom hook.

Reply:
hmmmmm I am still unsure about what is causing this issue. It seems a bit strange since I dont think that any other interfaces (e.g. whisper) are having this issue

Reply:
Indeed. I tried using the default behaviour but it just wouldn't work. So, I used the custom hooks as a work-around.